src/util/gobOption.ml

```diff
@@ -0,0 +1,9 @@
+open Prelude
+
+let exists p = function
+  | Some x -> p x
+  | None -> false
+
+let for_all p = function
+  | Some x -> p x
+  | None -> true
```

Comment:
These are very confusing, in what sense is this an `exists` or a `for_all`?

Reply:
In the sense that `option` is a singleton list or an empty list. This is a fairly standard generalization of `exists` and `for_all` to other data structures. For example Scala and Haskell have such operations for their optional types.